Automattic/woocommerce-payments
# Hide task list buttons

Fixes #2455.

#### Changes proposed in this Pull Request

This PR removes buttons from all task list items.

## Without UPE settings preview

<table>
<tr>
<th>Previously</th>
<th>Now</th>
<tr>
<td>

![Markup on 2021-08-05 at 10:34:35](https://user-images.githubusercontent.com/1759681/128320017-2a568f8a-10e2-41ef-bd25-4f3c6d8bbc25.png)

</td>
<td>

![Markup on 2021-08-06 at 08:56:03](https://user-images.githubusercontent.com/1759681/128470501-0aca304c-9ce8-4c4d-af8e-b313e0515f2b.png)

</td>
</tr>
</table>

## With UPE settings preview

Exactly the same as above side from the last item.

<table>
<tr>
<th>Previously</th>
<th>Now</th>
<tr>
<td>

![Markup on 2021-08-05 at 10:34:57](https://user-images.githubusercontent.com/1759681/128320001-526bd6ba-6eaa-468b-8aed-40646c3aa01e.png)

</td>
<td>

![Markup on 2021-08-06 at 08:56:46](https://user-images.githubusercontent.com/1759681/128470518-3c7dcadf-ec92-4bf1-ad42-7243ec3d3024.png)


</td>
</tr>
</table>

## Keyboard navigation demo

![Screen Capture on 2021-08-06 at 09-08-25](https://user-images.githubusercontent.com/1759681/128470781-9d76ca5c-fbdc-47fb-bdee-41e64f70d00b.gif)

#### Testing instructions





* In `WC_Payments_Admin::register_payments_scripts()`, find the line where `$wcpay_settings` is defined and just below it add this code to get all tasks to display:
  ```php
  $wcpay_settings['featureFlags']['accountOverviewTaskList'] = true;
  $wcpay_settings['accountStatus']['status'] = 'foo';
  $wcpay_settings['showUpdateDetailsTask'] = 'yes';
  $wcpay_settings['wpcomReconnectUrl'] = 'https://example.org';
  $wcpay_settings['needsHttpsSetup'] = true;
  $wcpay_settings['additionalMethodsSetup']['isTaskVisible'] = true;
  $wcpay_settings['additionalMethodsSetup']['isSetupCompleted'] = false;
  ```
* To test the "Without UPE preview" scenario:
  * Enable `_wcpay_feature_upe`.
  * Disable `_wcpay_feature_upe_settings_preview`.
* To test the "With UPE preview" scenario:
  * Enable `_wcpay_feature_upe_settings_preview`.
  * Disable `_wcpay_feature_upe`.
* Verify the tasks are displayed like in the "Now" screenshots.

-------------------

  - Not added - unreleased feature.
